### PR TITLE
Migrate array less_than operator to new framework

### LINKED ISSUE #4674: Migrate array less_than operator to new framework
See https://github.com/cberner/presto/commit/b0f94c381f61465779f909ed6592dfcd8fdbd01a for an example of how this is done. The code is in `ArrayLessThanOperator.java`

### PR DESCRIPTION
**Pull request by idemura**
Fixes #4674 .

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayLessThanOperator.java [hunk 1]
```diff
@@ -13,57 +13,37 @@
...
-        MethodHandle method = METHOD_HANDLE.bindTo(lessThanFunction).bindTo(elementType);
-        return new ScalarFunctionImplementation(false, ImmutableList.of(false, false), method, isDeterministic());
-    }
-
-    @UsedByGeneratedCode
-    public static boolean lessThan(MethodHandle lessThanFunction, Type type, Block leftArray, Block rightArray)
+    @TypeParameter("E")
+    @SqlType(StandardTypes.BOOLEAN)
+    public static boolean lessThan(
+                    @OperatorDependency(operator = LESS_THAN, returnType = StandardTypes.BOOLEAN, argumentTypes = {"E", "E"}) MethodHandle lessThanFunction,
```
Comment (haozhun) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayLessThanOperator.java]: We indent 4 for indent and 8 for continuation indent.

**Comment by haozhun** [comment]
Please address my comments.

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayLessThanOperator.java (hunk 1 above)
Comment (idemura) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayLessThanOperator.java]: Done

**Comment by idemura** [comment]
Updated change

**Comment by haozhun** [comment]
Merged #6002 after rebase. Thank you.

**Pull request closed without merging**